src/DynamoCore/Graph/Nodes/CodeBlockNode.cs

```diff
@@ -931,28 +958,45 @@ private void LoadAndCreateConnectors(OrderedDictionary inportConnections, Ordere
             List<int> undefinedIndices = new List<int>();
             for (int i = 0; i < OutPorts.Count; i++)
             {
+                // If a code block is in an error state the indexes are not always
+                // known (after the code block node is loaded in an error state), 
+                // so matching the connector by name can result in the port being 
+                // on the wrong line, just store the index to match in step 2 next
+                if (IsInErrorState)
+                {
+                    undefinedIndices.Add(i);
```

Comment:
this initial check seems to switch the priority for the different steps, does this alter runtime behavior much? For example, where you have a code block, you go into an error state, you modify it and then fix it, do connectors still reconnect semi accurately?

Reply:
This change is the main point of this fix. If the code is in an error state the names will match leading to using the old line indexes, which is incorrect. In the case of the code being in an error state this section of code needs to be explicitly skipped. With this section of code being run the connectors are incorrect. Without this section of code the connectors seem to function as expected, at least with the amount of testing that I have done so far.

Reply:
my point is that the names are unknown only when loading a code block in an error state - but I am curious if this degrades the behavior of the connector recreation for code blocks which were created at runtime (no saving or loading).

Reply:
I tested it by loading the error test file, and then fixing the error. All connections behaved as expected. After that I deleted the fix for the error (basically undoing without using undo), and the connections remained the same as they were after the fix, meaning that they did not revert back to the initial loaded state but remained connected. Re-fixing the error caused no apparent change in the connectors as the ports were already placed on the correct lines.

The one place where this change does cause a difference from the previous behavior is in the case where a code block node was in an error state and is then fixed to not be in an error state. In this case matching by name is disabled and matching by index takes priority. When loading a saved code block node in an error state this is what is needed to bypass the initial (incorrect) port locations. In other cases it is unclear if this is an issue or not. I have not found any cases where this change would cause an issue, but I have not tested this specific situation fully yet.

One suggestion could be that the "IsInErrorState" check could be expanded to test if the file is just first being loaded or not. Some testing will clarify further if this is needed, but so far it seems to work as expected for me.